x/pool-incentives/keeper/distr.go

```diff
@@ -69,13 +47,18 @@ func (k Keeper) AllocateAsset(ctx sdk.Context, asset sdk.Coin) error {
 	totalWeightDec := distrInfo.TotalWeight.ToDec()
 	for _, record := range distrInfo.Records {
 		allocatingAmount := assetAmountDec.Mul(record.Weight.ToDec().Quo(totalWeightDec)).TruncateInt()
-		coins := sdk.NewCoins(sdk.NewCoin(asset.Denom, allocatingAmount))
+
+		// when weight is too small and no amount is allocated, just skip this to avoid zero coin send issues
+		if !allocatingAmount.IsPositive() {
```

Comment:
If a pot runs out of rewards, does it get deleted unless its a perpetual pot? (And the only perpetual pot is what is created for OSMO rewards?)

Reply:
wdym by pools are perpetual here?
Can we log a warning to the console in this situation, just so we can know if it ever happened? This hopefully shouldn't happen, since the chain shouldn't allocate a low percentage to any single pool

Were iterating over distrInfo

Reply:
The automatically created Pots by pool creation  are perpetual pots I mean.
```
			ctx,
			true,
			k.accountKeeper.GetModuleAddress(types.ModuleName),
			sdk.Coins{},
			lockuptypes.QueryCondition{
				LockQueryType: lockuptypes.ByDuration,
				Denom:         gammtypes.GetPoolShareDenom(poolId),
				Duration:      lockableDuration,
				Timestamp:     time.Time{},
			},
			// QUESTION: Should we set the startTime as the epoch start time that the modules share or the current block time?
			ctx.BlockTime(),
			1,
```

Reply:
Btw, added logger to log the allocating amount zero event